src/systemctl/systemctl.c

```diff
@@ -5684,23 +5684,22 @@ static int add_dependency(int argc, char *argv[], void *userdata) {
 }
 
 static int preset_all(int argc, char *argv[], void *userdata) {
-        UnitFileChange *changes = NULL;
-        unsigned n_changes = 0;
         int r;
 
         if (install_client_side()) {
+                UnitFileChange *changes = NULL;
+                unsigned n_changes = 0;
 
                 r = unit_file_preset_all(arg_scope, arg_runtime, arg_root, arg_preset_mode, arg_force, &changes, &n_changes);
-                if (r < 0) {
+                if (r < 0)
                         log_error_errno(r, "Operation failed: %m");
-                        goto finish;
+                else {
```

Comment:
unless I misunderstand the code now: r gets eaten up here now, but should actually be returned.